Adding ZodZero split keyboard

Aleblazer opened the pull request:
## Description

Adding my split 40% ZodZero split keyboard with Default keymap

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

waffle87 commented on keyboards/aleblazer/zodzero/info.json:
```diff
@@ -0,0 +1,301 @@
+        "pid": "0xF911",
+        "device_version": "0.0.1"
+    },
+    "bootmagic": {
+        "matrix": [ 0, 0]
+    },
```
```suggestion
```

waffle87 commented on keyboards/aleblazer/zodzero/keymaps/aleblazer/config.h:
```diff
@@ -0,0 +1,75 @@
+     /* RGB Defines */
+#define RGB_DI_PIN GP0
+#define RGB_MATRIX_LED_COUNT 70
+#define RGBLED_NUM 70
```
```suggestion
```

waffle87 commented on keyboards/aleblazer/zodzero/keymaps/aleblazer/config.h:
```diff
@@ -0,0 +1,75 @@
+#define ENABLE_RGB_MATRIX_BAND_SAT
+#define ENABLE_RGB_MATRIX_BAND_VAL
+#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
+#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
+#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
+#define ENABLE_RGB_MATRIX_CYCLE_ALL
+#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
+#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
+#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
+#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
+#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
+#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
+#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
+#define ENABLE_RGB_MATRIX_DUAL_BEACON
+#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
+#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
+#define ENABLE_RGB_MATRIX_RAINDROPS
+#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
+#define ENABLE_RGB_MATRIX_HUE_BREATHING
+#define ENABLE_RGB_MATRIX_HUE_PENDULUM
+#define ENABLE_RGB_MATRIX_HUE_WAVE
+#define ENABLE_RGB_MATRIX_PIXEL_RAIN
+#define ENABLE_RGB_MATRIX_PIXEL_FLOW
+#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
+#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
+#define ENABLE_RGB_MATRIX_DIGITAL_RAIN
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
+#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
+#define ENABLE_RGB_MATRIX_SPLASH
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#endif
```
I've a feeling this should go at the keyboard level `config.h`? Assuming RGB and OLEDs are meant to be installed. Same goes for other configuration in `rules.mk`, `keymap.c`, etc. Then removed from the other keymaps as well.

Aleblazer replied:
Moved this to the keyboard level config.h

waffle87 commented on keyboards/aleblazer/zodzero/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,383 @@
+            tap_code(KC_VOLD);
+        }
+    }
+    return true;
```
```suggestion
    return false;
```
Although again, this should be moved to the keyboard level using the `encoder_update_kb` function.

Aleblazer replied:
Moved to zodzero.h under encoder_update_kb

waffle87 commented on keyboards/aleblazer/zodzero/readme.md:
```diff
@@ -0,0 +1,21 @@
+# ZodZero
+
+![ZodZero](https://i.imgur.com/iOvyZLM.jpeg)
```
```suggestion
![ZodZero](https://i.imgur.com/iOvyZLMh.jpeg)
```

waffle87 commented on keyboards/aleblazer/zodzero/readme.md:
```diff
@@ -0,0 +1,21 @@
+
+    make aleblazer/zodzero:default
+
+Flashing example for this keyboard:
```
```suggestion
Flashing example for this keyboard:

    make aleblazer/zodzero:default: flash
    
## Bootloader
```

Aleblazer replied:
Updated this.

waffle87 commented on keyboards/aleblazer/zodzero/zodzero.h:
```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include "quantum.h"
+
+#define LAYOUT( \
+L00, L01, L02, L03, L04, L05,           L37,   R30,           R02, R03, R04, R05, R06, R07, \
+L10, L11, L12, L13, L14, L15, L06, L07, L17,   R10, R00, R01, R12, R13, R14, R15, R16, R17, \
+L20, L21, L22, L23, L24, L25, L16, L26, L27,   R20, R21, R11, R22, R23, R24, R25, R26, R27, \
+L30, L31, L32, L33, L34,    L35,   L36,             R31,   R32,    R33, R34, R35, R36, R37  \
+  ) \
+  { \
+ { L00,   L01,   L02,   L03,   L04,   L05,   L06,   L07 }, \
+ { L10,   L11,   L12,   L13,   L14,   L15,   L16,   L17 }, \
+ { L20,   L21,   L22,   L23,   L24,   L25,   L26,   L27 }, \
+ { L30,   L31,   L32,   L33,   L34,   L35,   L36,   L37 }, \
+ { R07,   R06,   R05,   R04,   R03,   R02,   R01,   R00 }, \
+ { R17,   R16,   R15,   R14,   R13,   R12,   R11,   R10 }, \
+ { R27,   R26,   R25,   R24,   R23,   R22,   R21,   R20 }, \
+ { R37,   R36,   R35,   R34,   R33,   R32,   R31,   R30 }  \
+}
```
This belongs in `info.json`, and this file removed.

Aleblazer replied:
Updated info.json with the matrix and placement data, removed zodzero.h, and tested with new default keymap.

drashna commented on keyboards/aleblazer/zodzero/keymaps/default/keymap.c:
In general, we want to keep the default keymap simple and plain.  Ideally, it should just be the keymap array, and encoder map array (if it's enabled).

Aleblazer replied:
Moved all RGB Matrix, Encoder, and OLED code to the keyboard level config.h, rules.mk, and zodzero.c.

Aleblazer commented on keyboards/aleblazer/zodzero/keymaps/aleblazer/keymap.c:
```diff
@@ -0,0 +1,423 @@
+
+        oled_set_cursor(0, 0);                            // sets cursor to (row, column) using charactar spacing (5 rows on 128x32 screen, anything more will overflow back to the top)
+        sprintf(wpm_str, "WPM:%03d", get_current_wpm());  // edit the string to change wwhat shows up, edit %03d to change how many digits show up
+        oled_write(wpm_str, false);                       // writes wpm on top left corner of string
```
Removed this keymap.

Aleblazer commented on keyboards/aleblazer/zodzero/keymaps/default/config.h:
```diff
@@ -0,0 +1,73 @@
+#define ENABLE_RGB_MATRIX_MULTISPLASH
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+#endif
```
Moved this config to the keyboard level config.h

Aleblazer commented on keyboards/aleblazer/zodzero/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,383 @@
+                layer_off(_RAISE);
+                update_tri_layer(_LOWER, _RAISE, _ADJUST);
+            }
+            return false;
```
Removed this portion of the keymap, using simple layers now.

Aleblazer commented on keyboards/aleblazer/zodzero/keymaps/default/rules.mk:
```diff
@@ -0,0 +1,7 @@
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = vendor
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
```
Moved this config to the keyboard level.

Aleblazer commented on keyboards/aleblazer/zodzero/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,383 @@
+    ),
+};
+
+#ifdef OLED_ENABLE
```
Moved to zodzero.c, and changed oled_task_user to oled_task_kb.

Aleblazer commented:
So to avoid having to rewrite some legacy layer code I typically use, I removed the aleblazer keymap, and on the default, removed the custom layers/keycodes, and just used simple numbered layers. Moved all config that was in the keymap level to the keyboard level, and moved the OLED and Encoder code to the keyboard level zodzero.c. Updated the info.json with the correct positioning, and the matrix data. Deleted zodzero.h.

drashna commented on keyboards/aleblazer/zodzero/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,54 @@
+*/
+
+#include QMK_KEYBOARD_H
+#include <stdio.h>
+bool is_alt_tab_active = false;
+uint16_t alt_tab_timer = 0;
+
+
```
```suggestion
```

drashna commented on keyboards/aleblazer/zodzero/zodzero.c:
```diff
@@ -0,0 +1,266 @@
+    return rotation;
+}
+
+bool oled_task_kb(void) {
```
```suggestion
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
```

Aleblazer commented:
@drashna and @waffle87, just checking on if there are any pending reviews needed for this pull request, I believe all request have been addressed.

drashna commented on keyboards/aleblazer/zodzero/rules.mk:
```diff
@@ -0,0 +1,9 @@
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = vendor
+OLED_ENABLE = yes
+OLED_DRIVER = SSD1306
```
This is the default, and isn't needed.
```suggestion
```

fauxpark commented on keyboards/aleblazer/zodzero/readme.md:
```diff
@@ -0,0 +1,22 @@
+* Hardware Supported: RP2040-Zero
+* Hardware Availability: [Split Logic Keyboards](https://splitlogic.xyz/)
+
+## Make example for this keyboard (after setting up your build environment):
+
+    make aleblazer/zodzero:default
+
+## Flashing example for this keyboard:
+    make aleblazer/zodzero:default: flash
```
```suggestion
Make example for this keyboard (after setting up your build environment):

    make aleblazer/zodzero:default

Flashing example for this keyboard:

    make aleblazer/zodzero:default:flash
```

fauxpark commented on keyboards/aleblazer/zodzero/config.h:
```diff
@@ -0,0 +1,84 @@
+
+#ifdef RGB_MATRIX_ENABLE
+     /* RGB Defines */
+#define WS2812_DI_PIN GP0
```
```suggestion
```

fauxpark commented on keyboards/aleblazer/zodzero/info.json:
```diff
@@ -0,0 +1,104 @@
+        "vid": "0xF901",
+        "pid": "0xF911",
+        "device_version": "0.0.1"
+    },
```
```suggestion
    },
    "rgb_matrix": {
        "driver": "ws2812"
    },
    "ws2812": {
        "driver": "vendor",
        "pin": "GP0"
    },
```

fauxpark commented on keyboards/aleblazer/zodzero/rules.mk:
```diff
@@ -0,0 +1,9 @@
+MOUSEKEY_ENABLE  = yes
+EXTRAKEY_ENABLE  = yes
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+WS2812_DRIVER = vendor
```
```suggestion
```
